### PR TITLE
gunicorn appid improvement

### PR DESCRIPTION
**Pull request by pfilipko1**
Works properly, not sure about the naming convention though.
https://github.com/Granulate/gprofiler/issues/704

### CODE REVIEW

**Review by Jongy — CHANGES_REQUESTED** [review]
Before I begin reviewing, can you please add tests next to `test_gunicorn` in `test_appids.py`? (tests that show this change now supports the cases described in the ticket)

**Comment by pfilipko1** [comment]
Added two test cases (hopefully correct) containing commands with multiple arguments containing ":"

File: gprofiler/metadata/application_identifiers.py [hunk 1]
```diff
@@ -116,8 +126,8 @@ def get_app_id(self, process: Process) -> Optional[str]:
         ) and "gunicorn" != os.path.basename(_get_cli_arg_by_index(process.cmdline(), 1)):
             return None
 
-        # wsgi app specification will come always as the last argument (if hasn't been specified config file)
-        return self.gunicorn_to_app_id(process.cmdline()[-1], process)
+        # wsgi app specification must not be the last argument, thus the method to extract it
```
Comment (Jongy) [inline gprofiler/metadata/application_identifiers.py:129]:
```suggestion
        # wsgi app specification might not be the last argument, thus the method to extract it
```

File: gprofiler/metadata/application_identifiers.py [hunk 2]
```diff
@@ -104,6 +104,16 @@ def gunicorn_to_app_id(self, wsgi_app_spec: str, process: Process) -> str:
...
         return f"gunicorn: {wsgi_app_spec} ({_append_python_module_to_proc_wd(process, wsgi_app_file)})"
 
+    def gunicorn_get_app_name(self, cmdline_list: List[str]) -> str:
+        # method improving appid selection:
+        # https://github.com/Granulate/gprofiler/issues/704
+        cmdline_colon_list = list(filter(lambda x: ":" in x, cmdline_list))
+        for index, arg in enumerate(cmdline_list):
+            for colon_arg in cmdline_colon_list:
+                if arg == colon_arg and not cmdline_list[index - 1].startswith("-"):
+                    return arg
```
Comment (Jongy) [inline gprofiler/metadata/application_identifiers.py:114]: Why do you require `cmdline_colon_list`? Can't you replace `arg == colon_arg` with `":" in arg`?

File: tests/test_appids.py
Comment (Jongy) [inline tests/test_appids.py:1]: Let's add one test case for the negative case, i.e for `return "unknown app name"`.

**Review by Jongy — CHANGES_REQUESTED** [review]
.

**Review by Jongy — APPROVED** [review]
